### DISCUSSION TITLE
Matrix writer for pixel values in shader

### DISCUSSION THREAD

**Discussion by lblabucsd** [bonsai-rx/bonsai#1286] [Q&A]
Hi Bonsai friends, 
I'd like to ask for help in order to understand how to create the best numerical description of my shader content. 
I load a texture (meaning a jpg) to create a visual environment in a shader window. Since during an experiment this texture is being moved in closed-loop, I want to document where exactly the image components of my texture can be found in the shader. Ideally, I'd like to have a matrix for all pixel values and data points.
I tried to "ReadPixels" from the shader video and "CsvWriter" from there. However, this only gave me a chart with booleans and a bunch of fix values.
Thanks a lot for your ideas and best, 
Lisanne

**Comment by bruno-f-cruz** [comment]
In order to log the output of an IplImage use ImageWriter, VideoWriter (probably with a raw codec) or even MatrixWriter.

I would just like to point out that it is usually better to save some sort of low-level description of the stim parameters (such as X,Y location of the center of the texture) than to save the whole shader render. This operation might destabilize your rendering pipeline and lead to stuttering for instance.

Hope it helps,
Bruno
